test/node/features/fixtures/koa/scenarios/app.js

```diff
@@ -59,6 +57,21 @@ app.use(async (ctx, next) => {
     await next()
   } else if (ctx.path === '/bodytest') {
     throw new Error('request body')
+  } else if (ctx.path === '/throw-async-callback') {
+    setTimeout(function () {
+      throw new Error('error in async callback')
+    }, 100)
+    ctx.body = 'ok'
+  } else if (ctx.path === '/reject-async-callback') {
+    setTimeout(function () {
+      Promise.reject(new Error('reject in async callback')).catch(next)
+    }, 100)
+    ctx.body = 'ok'
+  } else if (ctx.path === '/unhandled-reject-async-callback') {
+    setTimeout(function () {
+      Promise.reject(new Error('unhandled reject in async callback'))
```

Comment:
promise rejections are not caught by the domain 'error' handler so still to go `unhandledPromiseRejection`